test/test_utils/ec2.py

```diff
@@ -459,6 +460,47 @@ def get_ec2_instance_tags(instance_id, region=DEFAULT_REGION, ec2_client=None):
     return {tag["Key"]: tag["Value"] for tag in response.get("Tags")}
 
 
+def enforce_IMDSv2(instance_id, region=DEFAULT_REGION, ec2_client=None, hop_limit = 1):
+    """
+    Enabled HTTP TOKENS required option on EC2 instance with given hop limit.
+    
+    :param instance_id: str, ec2 instance id
+    :param region: str, Region where ec2 instance is launched.
+    :param ec2_client: str, ec2 client.
+    :param hop_limit: str, hop limit to be set on ec2 instance.
+    """
+    ec2_client = ec2_client or get_ec2_client(region)
+    IMDSv2_enforced = False
+    response = ec2_client.modify_instance_metadata_options(
+        InstanceId=instance_id,
+        HttpTokens='required',
+        HttpPutResponseHopLimit=hop_limit,
+        HttpEndpoint='enabled',
+    )
+
+    if not response:
+        raise Exception("Unable to enforce IMDSv2. No response received.")
+
+    timeout = 3
+    state = None
+    if response["InstanceId"]:
```

Comment:
May cause unexpected KeyError if InstanceId not in response. Might be beneficial to use response.get("InstanceId") to avoid unexpected errors